lib/logstash/outputs/elasticsearch.rb

```diff
@@ -263,6 +265,10 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   config :user, :validate => :string
   config :password, :validate => :password
 
+  # HTTP Path at which the Elasticsearch server lives. Use this if you must run ES behind a proxy that remaps
+  # the root path for the Elasticsearch HTTP API lives. This option is ignored for non-HTTP transports.
+  config :path, :validate => :string, :default => "/"
```

Comment:
I agree that it may be inconvenient for folks just trying different transports. It's a bit of a pickle because we have many settings that don't work on differet protocols (bind_host/port isn't used in transport or http, for example).

It can stay how it is, and we can address it further with warnings or errors if users get confused by it. +1 leaving as is.